qiskit_addon_dice_solver/dice_solver.py

```diff
@@ -46,6 +47,124 @@ def __init__(self, command, returncode, log_path):
+            will be projected and diagonalized. This is a 2D array of ``bool`` representations of bit
+            values such that each row represents a single bitstring. The spin-up configurations
+            should be specified by column indices in range ``(N, N/2]``, and the spin-down
+            configurations should be specified by column indices in range ``(N/2, 0]``, where ``N``
+            is the number of qubits.
+        hcore: Core Hamiltonian matrix representing single-electron integrals
+        eri: Electronic repulsion integrals representing two-electron integrals
+        mpirun_options: Options controlling the CPU resource allocation for the ``Dice`` command line application.
+            These command-line options will be passed directly to the ``mpirun`` command line application during
+            invocation of ``Dice``. These may be formatted as a ``Sequence`` of strings or a single string. If a ``Sequence``,
+            the elements will be combined into a single, space-delimited string and passed to
+            ``mpirun``. If the input is a single string, it will be passed to ``mpirun`` as-is. If no
+            ``mpirun_options`` are provided by the user, ``Dice`` will run on a single MPI slot. For more
+            information on the ``mpirun`` command line options, refer to the `man page <https://www.open-mpi.org/doc/current/man1/mpirun.1.php>`_.
+        working_dir: An absolute path to a directory for storing temporary files. If
+            not provided, the system temporary files directory will be used.
+        clean_working_dir: Whether to delete intermediate files generated by the ``Dice`` command line application.
+            These files will be stored in a directory created inside ``working_dir``.
+            If ``False``, then this directory will be preserved.
+
+    Returns:
+        - Minimum energy from SCI calculation
+        - SCI coefficients
+        - Average orbital occupancy
+    """
+    # Hard-code target S^2 until supported
+    spin_sq = 0.0
+
+    # Convert bitstring matrix to integer determinants for spin-up/down
+    ci_strs = bitstring_matrix_to_ci_strs(bitstring_matrix)
+    num_configurations = len(ci_strs[0])
+    num_up = bin(ci_strs[0][0])[2:].count("1")
+    num_dn = bin(ci_strs[1][0])[2:].count("1")
+
+    # Set up the working directory
+    working_dir = working_dir or tempfile.gettempdir()
+    intermediate_dir = Path(tempfile.mkdtemp(prefix="dice_cli_files_", dir=working_dir))
+
+    # Write the integrals out as an FCI dump for Dice command line app
+    active_space_path = os.path.join(intermediate_dir, "fcidump.txt")
+    num_orbitals = hcore.shape[0]
+    tools.fcidump.from_integrals(
+        active_space_path, hcore, eri, num_orbitals, (num_up + num_dn)
+    )
+
+    _write_input_files(
+        ci_strs,
+        active_space_path,
+        num_up,
+        num_dn,
+        num_configurations,
+        intermediate_dir,
+        spin_sq,
+        1,
```

Comment:
For readability, please use keyword arguments for this function call.

Reply:
Done, let's go with `dice_dir` for the dice I/O and `temp_dir` for the input directory in which we create these `dice_dirs`

Will clean up "addresses" lingo in a separate PR throughout module, with exception of `solve_dice` for now